app/workers/workarea/update_product_bundles.rb

```diff
@@ -0,0 +1,26 @@
+module Workarea
+  class UpdateProductBundles
+    include Sidekiq::Worker
+    include Sidekiq::CallbacksWorker
+
+    sidekiq_options(
+      enqueue_on: { Catalog::Product => :destroy },
+      unique: :until_executing
+    )
+
+    def perform(id)
+      Sidekiq::Callbacks.inline do
+        bundles = Catalog::Product.where(product_ids: id)
+        bundles.each do |bundle|
+          bundle.product_ids.delete(id)
```

Comment:
So is this just removing the element from the array? Does that get persisted? I've had Mongoid trouble recently with mutating without using the setter. `bundle.product_ids -= [id]`

Reply:
yes, and yes. While it looks like it because I changed the name of the class and added the variant components stuff, this isn't new code. So it has been working for a while.

Reply:
Ok cool. It's not localized, so that's probably why it works.